src/SelfService/Application/PlatformDataApiRequesterService.cs

```diff
@@ -126,16 +192,51 @@ public async Task<MyCapabilityCosts> GetMyCapabilitiesCosts(UserId userId, int d
         return new MyCapabilityCosts(costs);
     }
 
-    public async Task<CapabilityCosts> GetCapabilityCosts(CapabilityId capabilityId, int daysWindow)
+    public async Task<MyCapabilitiesAwsResourceCounts> GetMyCapabilitiesAwsResourceCounts(UserId userId)
     {
-        var timeSeriesWithValidCapabilities = await FetchCapabilityCosts(
-            new(QueryParamCapabilityId, capabilityId),
-            new(QueryParamDaysWindow, daysWindow.ToString())
-        );
+        var myCapabilities = await _myCapabilitiesQuery.FindBy(userId);
+        var capabilitiesArray = myCapabilities as Capability[] ?? myCapabilities.ToArray();
+        var awsAccountIdToCapabilityIdMap = new Dictionary<string, CapabilityId>();
+        foreach (var myCapability in capabilitiesArray)
+        {
+            var awsAccount = await _awsAccountRepository.FindBy(myCapability.Id);
+            if (awsAccount?.Registration.AccountId is null)
+                continue;
+            var id = awsAccount.Registration.AccountId?.ToString();
+            awsAccountIdToCapabilityIdMap.Add(id!, myCapability.Id);
+        }
 
-        var mappedCosts = ToCapabilityMap(timeSeriesWithValidCapabilities);
-        return mappedCosts.TryGetValue(capabilityId, out var cost)
-            ? new CapabilityCosts(capabilityId, cost.ToArray())
-            : new CapabilityCosts(capabilityId, new TimeSeries[] { });
+        var awsResourceCounts = await FetchAwsResourceCounts();
+
+        var mappedCounts = ToAwsAccountMap(awsResourceCounts);
+
+        List<CapabilityAwsResourceCounts> resourceCounts = new List<CapabilityAwsResourceCounts>();
+        foreach (var accountsCount in awsResourceCounts)
+        {
+            if (!awsAccountIdToCapabilityIdMap.TryGetValue(accountsCount.AwsAccountId, out var capabilityId))
+            {
+                // not an error, remember my capabilities are a subset of all capabilities
+                continue;
+            }
+
+            if (!mappedCounts.TryGetValue(accountsCount.AwsAccountId, out var counts))
```

Comment:
How does this even happen?
They are based on the same source data, are they not?

Reply:
Discussed on slack. This is fine.